### PR TITLE
Initial docs generation

### PR DESCRIPTION
**Pull request by ppdebreuck**
Modnet documentation automatically generated by sphinx apidoc.

To be discussed:
- What files should exactly be tracked by GitHub ? (everything has bee committed)
- Is the overall structure of the docs understandable, make it more fancy (some extra sections)?
- Host it somewhere, preferentially automatically generated and updated

### CODE REVIEW

**Comment by davidwaroquiers** [comment]
Hi guys,

Thank you for the hard work you've been doing with MODNet, which I haven't been able to follow up very much. Just one general comment on documentation, it would be good to get off on the right foot from the beginning (of course :D). I've seen this blog and there is a nice talk from the guy at a pycon conference: [https://documentation.divio.com/](https://documentation.divio.com/)

Everything does not need to be there from the beginning but if the structure is laid out nicely from the beginning it will for sure help for the future.

David

**Comment by ml-evs** [comment]
Just to put my suggestions to @ppdebreuck in public:

- we could use readthedocs to host the sphinx docs, that would be built on every push to master. It allows you to have versioned documentation which is nice.
- we could "self-host" on github pages, by committing the docs to a `gh_pages` branch in this repository (and hitting a switch in the settings); we could then have a CI task that rebuilds the docs and commits them to the `gh_pages` branch automatically (so only the build files would need to be committed to master)

**Comment by ppdebreuck** [comment]
Update:
- Added other sections (mainly Install + Tutorials). Will look how to integrate the other sections recommended by David.
- individual pages for MODData and MODNetModel's specifications
- Only track source files+ makefile
- Hosting will be done on ReadtheDocs
- Included README update in this PR

File: README.md [hunk 1]
```diff
@@ -19,9 +20,13 @@
 
 <a name="introduction"></a>
 ## Introduction
-This repository contains the python package implementing the Material Optimal Descriptor Network (MODNet). It is a supervised machine learning framework for **learning material properties** from the **crystal structure**. The framework is well suited for **limited datasets** and can be used for learning *multiple* properties together by using **joint transfer learning**.
+This repository contains the python package implementing the Material Optimal Descriptor Network (MODNet).
```
Comment (ml-evs) [inline README.md:23]:
```suggestion
This repository contains the Python (3.8+) package implementing the Material Optimal Descriptor Network (MODNet).
```

File: README.md [hunk 2]
```diff
@@ -19,9 +20,13 @@
...
+It is a supervised machine learning framework for **learning material properties** from
+either the **composition** or  **crystal structure**. The framework is well suited for **limited datasets**
+and can be used for learning *multiple* properties together by using **joint transfer learning**.
```
Comment (ml-evs) [inline README.md:26]: I would suggest that joint and transfer learning are two separate techniques?

```suggestion
and can be used for learning *multiple* properties together by using **joint learning**.
```

File: README.md [hunk 3]
```diff
@@ -19,9 +20,13 @@
...
 
-This repository also contains two pretrained models that can be used for predicting the refractive index and vibrational thermodynamics for any crystal structure.
+This repository also contains two pretrained models that can be used for predicting
+the refractive index and vibrational thermodynamics from any crystal structure.
```
Comment (ml-evs) [inline README.md:29]: (Just removing the linebreak)

```suggestion
This repository also contains two pretrained models that can be used for predicting the refractive index and vibrational thermodynamics from any crystal structure.
```

File: README.md [hunk 4]
```diff
@@ -34,7 +39,7 @@ See the MODNet papers and repositories below for more details:
 
 
 <p align='center'>
-<img src="img/MODNet_schematic.PNG" alt="MODNet schematic" />
+<img src="img/MODNet_schematic.png" alt="MODNet schematic" />
```
Comment (ml-evs) [inline README.md:42]: This image does not render for me when I "view file"

File: README.md [hunk 5]
```diff
@@ -113,16 +123,23 @@ data_MP = MODData.load('path/to/moddata/MP_2018.6')
 
 ```
 
-The latter MODData (MP_2018.6) is very usefull for predicting a learned property on all structures from the Materials Project:
+The MP MODData on [figshare](https://figshare.com/articles/dataset/Materials_Project_MP_2018_6_MODData/12834275)
+(MP_2018.6) is very usefull for predicting a learned property on all structures from the Materials Project:
```
Comment (ml-evs) [inline README.md:127]: Do we release the MP model too? The example line then shows model predictions...

```suggestion
The MP MODData on [figshare](https://figshare.com/articles/dataset/Materials_Project_MP_2018_6_MODData/12834275) (MP_2018.6) is very useful for predicting a learned property on all structures from the Materials Project:
```

File: README.md [hunk 6]
```diff
@@ -113,16 +123,23 @@ data_MP = MODData.load('path/to/moddata/MP_2018.6')
...
 predictions_on_MP = model.predict(data_MP)
 ```
 
 <a name="documentation"></a>
 ## Documentation
-The two main classes, `MODData` and `MODNetModel`, are detailed here.
+Please read to docstrings for all methods and classes (docs coming soon).
 
+Especially, carefully read the two main classes, `MODData` and `MODNetModel` found in preprocessing and models modules.
+A `MODData` instance is used for representing a particular dataset. It contains a list of structures and corresponding properties:
```
Comment (ml-evs) [inline README.md:138]:
```suggestion
A `MODData` instance is used for representing a particular dataset. It contains a list of structures and corresponding properties.
```

File: README.md [hunk 7]
```diff
@@ -173,7 +197,7 @@ from modnet.preprocessing import MODData
 data = MODData.load('path/dataname')
 ```
 
-Both the save and load methods use pandas `.read_pickle(...)` and `.load_pickle(...)` which will try to compress/decompress files according to their file extensions (e.g. `".zip"`, `".tgz"` and `".bz2"`.
+Both the save and load methods use pandas `.read_pickle(...)` and `.load_pickle(...)` which will try to compress/decompress files according to their file extensions (e.g. `".zip"`, `".tgz"` and `".bz2").
```
Comment (ml-evs) [inline README.md:200]:
```suggestion
Both the save and load methods use pandas `.read_pickle(...)` and `.load_pickle(...)` which will try to compress/decompress files according to their file extensions (e.g. `".zip"`, `".tgz"` and `".bz2").
Please note that this method is **unsafe** can load arbitrary Python objects. Care has been taken to check the hashes of data that is automatically loaded from Figshare; if you rely on this feature for your own data then we recommend you do the same.
```

File: README.md [hunk 8]
```diff
@@ -205,36 +229,56 @@ The model is created by a MODNetModel instance:
...
 from modnet.models import MODNetModel
 
-model = MODNetModel(targets,weights,num_neurons=[[64],[32],[16],[16]], n_feat=300, loss='mse',act='relu')
+model = MODNetModel(targets,
+                    weights,
+                    num_classes = None  # only needed for classification, e.g. num_classes = {'is_metal':2}
+                    num_neurons=[[64],[32],[16],[16]],
+                    n_feat=300,
+                    act='relu',
+                    )
```
Comment (ml-evs) [inline README.md:238]:
```suggestion
model = MODNetModel(
    targets,
    weights,
    num_classes = None  # only needed for classification, e.g. num_classes = {'is_metal':2}
    num_neurons=[[64],[32],[16],[16]],
    n_feat=300,
    act='relu',
)
```

File: README.md [hunk 9]
```diff
@@ -205,36 +229,56 @@ The model is created by a MODNetModel instance:
...
+model.fit(data,
+          val_fraction = 0.0,
+          val_data = None,
+          val_key = None,
+          lr=0.001,
+          epochs = 200,
+          batch_size = 128,
+          xscale='minmax',
+          loss='mse',
+          )
```
Comment (ml-evs) [inline README.md:268]:
```suggestion
model.fit(
    data,
    val_fraction = 0.0,
    val_data = None,
    val_key = None,
    lr=0.001,
    epochs = 200,
    batch_size = 128,
    xscale='minmax',
    loss='mse',
)
```

File: README.md (hunk 4 above)
Comment (ml-evs) [reply README.md:42]: (just need to either revert this change or fix the image file extension)
